### ISSUE TITLE
.pack() spoils the list in its argumets

### ISSUE THREAD

**Issue by Kukarism** [aleaxit/gmpy#108]
gmpy 2.0.7
Python 3.4.2 Amd-64 (windows 7-64)
gmpy2.pack() spoils the list in its argumets. And, what dazzled me especially deeply, it also spoils some of the copies of original list. Not just named links to original list, but totally different copies with distinct names and content.

This error arises rarely. Consider the following code:

```
import gmpy2
from gmpy2 import mpz
import copy

def print_array(parNotes, parArray, parBase=None) :
	print(parNotes)
	if parBase == None : parBase = 16
	for r in range(len(parArray)) :
		if type(parArray[r]).__name__ == "mpq" :
			print("num=", hex(parArray[r].numerator).rjust(77))
			print("den=", hex(parArray[r].denominator).rjust(77))
			print()
		elif type(parArray[r]).__name__ == "mpz" :
			print(parArray[r].digits(parBase).rjust(77))
		else :
			print(hex(parArray[r]).rjust(77))

num1 = mpz(0x3be270e2e2cc865b9790e46f061972675da0d7d2c6edf0ce8277b93b)

dif = gmpy2.unpack(num1, 16)
dif_copy_list = list(dif) # distinct copy by list()
dif_copy_list[0] = 1 #to show that this is really another memory area/object tan original 'dif' list
dif_copy_slice = dif[:] # distinct copy by slice
dif_copy_slice[0] = 2 #to show that this is really another memory area/object tan original 'dif' list
dif_copy_copy = copy.copy(dif) # distinct copy by copy.copy()
dif_copy_copy[0] = 3 #to show that this is really another memory area/object tan original 'dif' list
dif_copy_deepcopy = copy.deepcopy(dif) # distinct copy by copy.copy()

print_array("dif", dif)
print_array("dif_copy_list", dif_copy_list)
print_array("dif_copy_slice", dif_copy_slice)
print_array("dif_copy_copy", dif_copy_copy)
print_array("dif_copy_deepcopy", dif_copy_deepcopy)
print("we have 5 different memory areas/lists with different names")

print("now spoiling just 1 original array 'dif' by packing into num2")
num2 = gmpy2.pack(dif, 16)
print("num1==num2", num1==num2)

print_array("dif", dif)
print_array("dif_copy_list", dif_copy_list)
print_array("dif_copy_slice", dif_copy_slice)
print_array("dif_copy_copy", dif_copy_copy)
print_array("dif_copy_deepcopy", dif_copy_deepcopy)

print("And now all the distinct memory areas - differently named copies of original list - ARE SPOILED ALSO, except the copy.deepcopy()")

```

**Comment by casevh** [comment]
Thanks for the detailed bug report. I understand the cause. I should have a fix in a few days.

Case

**Comment by casevh** [comment]
I have released gmpy2 2.0.8 which fixes the bug. Sorry about the delay.
